File: 151004/Terebey/kafka/discussion/src/main/java/org/education/bean/dto/MessageRequestTo.java
package org.education.bean.dto;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonGetter;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
public class MessageRequestTo {
    @JsonAlias("id")
    int id;

    @NotNull
    @Min(value = 0)
    Integer issueId;

    @NotNull
    @Size(min = 2, max = 2048, message = "Incorrect content size")
    String content;

    @JsonGetter("id")
    public int getId() {
        return id;
    }
}

File: 151004/Afanasenko/appLab2/src/main/kotlin/com/github/kornet_by/dc/lab2/dto/request/MessageRequestTo.kt
package com.github.kornet_by.dc.lab2.dto.request

import com.github.kornet_by.dc.lab2.bean.Message
import kotlinx.serialization.Serializable

@Serializable
data class MessageRequestTo(
	private val issueId: Long, private val content: String
) {
	fun toBean(id: Long?): Message = Message(
		id, issueId, content
	)

	init {
		require(content.length in 4..2048)
	}
}